devicetypes/smartthings/fibaro-smoke-sensor.src/fibaro-smoke-sensor.groovy

```diff
@@ -0,0 +1,453 @@
+        result << createEvent(descriptionText: cmd.toString(), isStateChange: false)
+    }
+    result
+}
+
+def smokeAlarmEvent(value) {
+    log.debug "smokeAlarmEvent(value): $value"
+    def map = [name: "smoke"]
+    if (value == 1 || value == 2) {
+        map.value = "detected"
+        map.descriptionText = "$device.displayName detected smoke"
+    } else if (value == 0) {
+        map.value = "clear"
+        map.descriptionText = "$device.displayName is clear (no smoke)"
+    } else if (value == 3) {
+        map.value = "tested"
+        map.descriptionText = "$device.displayName smoke alarm test"
+    } else if (value == 4) {
+        map.value = "replacement required"
+        map.descriptionText = "$device.displayName replacement required"
+    } else {
+        map.value = "unknown"
+        map.descriptionText = "$device.displayName unknown event"
+    }
+    createEvent(map)
+}
+
+def heatAlarmEvent(value) {
+    log.debug "heatAlarmEvent(value): $value"
+    def map = [name: "heatAlarm"]
+    if (value == 1 || value == 2) {
+        map.value = "overheat detected"
+        map.descriptionText = "$device.displayName overheat detected"
+    } else if (value == 0) {
+        map.value = "clear"
+        map.descriptionText = "$device.displayName heat alarm cleared (no overheat)"
+    } else if (value == 3 || value == 4) {
+        map.value = "rapid temperature rise"
+        map.descriptionText = "$device.displayName rapid temperature rise"
+    } else if (value == 5 || value == 6) {
+        map.value = "underheat detected"
+        map.descriptionText = "$device.displayName underheat detected"
+    } else {
+        map.value = "unknown"
+        map.descriptionText = "$device.displayName unknown event"
+    }
+    createEvent(map)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.wakeupv1.WakeUpNotification cmd) {
+    log.info "Executing zwaveEvent 84 (WakeUpV1): 07 (WakeUpNotification) with cmd: $cmd"
+    log.info "checking this MSR : ${getDataValue("MSR")} before sending configuration to device"
+    def result = [createEvent(descriptionText: "${device.displayName} woke up", isStateChange: false)]
+    /* check MSR = "manufacturerId-productTypeId" to make sure configuration commands are sent to the right model */
+    if (!isConfigured() && getDataValue("MSR")?.startsWith("010F-0C02")) {
+        result << response(configure()) // configure a newly joined device or joined device with preference update
+    } else {
+        //Only ask for battery if we havn't had a BatteryReport in a while
+        if (!state.lastbatt || (new Date().time) - state.lastbatt > 24*60*60*1000) {
+            log.debug("Device has been configured sending >> batteryGet()")
+            result << response(zwave.securityV1.securityMessageEncapsulation().encapsulate(zwave.batteryV1.batteryGet()).format())
+            result << response("delay 1200")  // leave time for device to respond to batteryGet
```

Comment:
If I recall correctly, this delay isn't going to work because it's not in the same HubAction as the other commands. Because of how these delays are processed, you have to get all of the commands in the same call to response() to make sure they'll be considered one command list:

```
cmds << zwave.security...format()
cmds << "delay 1200"
...
cmds << zwave.wakeUp...format()
result += response(cmds)
```

Maybe that issue has been fixed, not sure, but I've run into it a couple times before.